src/helpers/EnvironmentInfoHelper.ts

```diff
@@ -0,0 +1,59 @@
+import bowser from "bowser";
+import {EnvironmentInfo} from '../models/EnvironmentInfo';
+import BROWSER_TYPES from '../utils/BrowserTypes';
+import { OneSignalUtils } from '../utils/OneSignalUtils';
+
+export class EnvironmentInfoHelper {
+    public static getEnvironmentInfo() : EnvironmentInfo {
+        return {
+            isBrowser: this.isBrowser(),
+            browserType: this.getBrowser(),
+            browserVersion: this.getBrowserVersion(),
+            isHttps: this.isHttps(),
+            isUsingSubscriptionWorkaround: this.isUsingSubscriptionWorkaround(),
+            isBrowserAndSupportsServiceWorkers: this.isBrowserAndSupportsServiceWorkers(),
+            requiresUserInteraction: this.requiresUserInteraction(),
+            osVersion: this.getOsVersion()
+        }
+    }
+
+    private static isBrowser(): boolean {
+        return !!this.getBrowser();
+    }
+
+    private static getBrowser(): string {
+        // from bowser source: https://bit.ly/36mq1R5
+        return (BROWSER_TYPES as any)[bowser.name];
+    }
+
+    private static getBrowserVersion(): number {
+        return Number(bowser.version);
+    }
+
+    private static isHttps(): boolean {
+        return location.protocol == 'https:';
+    }
+
+    private static isUsingSubscriptionWorkaround(): boolean {
+        return OneSignalUtils.isUsingSubscriptionWorkaround();
+    }
+    
+    private static isBrowserAndSupportsServiceWorkers(): boolean {
+        return (this.isBrowser() && navigator && 'serviceWorker' in navigator);
+    }
+
+    private static requiresUserInteraction(): boolean {
+        var requiresUserInteraction = true;
```

Comment:
Default should be `false` by default. Only Safari 12.1+ and Firefox 72+ require user interaction to prompt.

Reply:
We could also clean up some of the checks we have for this now in production code.